compiler/rustc_codegen_llvm/src/debuginfo/metadata/enums/cpp_like.rs

```diff
@@ -532,21 +532,39 @@ fn build_variant_struct_wrapper_type_di_node<'ll, 'tcx>(
                 variant_struct_type_di_node,
             ));
 
-            let build_assoc_const =
-                |name: &str, type_di_node: &'ll DIType, value: u64, align: Align| unsafe {
-                    llvm::LLVMRustDIBuilderCreateStaticMemberType(
-                        DIB(cx),
-                        wrapper_struct_type_di_node,
-                        name.as_c_char_ptr(),
-                        name.len(),
-                        unknown_file_metadata(cx),
-                        UNKNOWN_LINE_NUMBER,
-                        type_di_node,
-                        DIFlags::FlagZero,
-                        Some(cx.const_u64(value)),
-                        align.bits() as u32,
-                    )
+            let build_assoc_const = |name: &str,
+                                     type_di_node_: &'ll DIType,
+                                     value: u64,
+                                     align: Align| unsafe {
+                // FIXME: Currently we force all DISCR_* values to be u64's as LLDB seems to have
+                // problems inspecting other value types. Since DISCR_* is typically only going to be
+                // directly inspected via the debugger visualizer - which compares it to the `tag` value
+                // (whose type is not modified at all) it shouldn't cause any real problems.
```

Comment:
can we file a bug for this against lldb? I think they don't have a separate repo from llvm, so it would just be against https://github.com/llvm/llvm-project

Reply:
i've opened an issue for it https://github.com/llvm/llvm-project/issues/119101